### PR TITLE
Api improvements

### LINKED ISSUE #322: Make `middle` in `boxes` and `y` in `crossBars` optional
They are now mandatory, however Lets-Plot may not display them if it's needed.

### LINKED ISSUE #334: Add line type parameter to custom style elements
https://github.com/JetBrains/lets-plot-kotlin/issues/235

### LINKED ISSUE #385: `spacerWidth` setting missing in pie charts
Lets-plot supports the `spacer_width` setting in pie charts to define, well, the spacer width. This can prevent issues like this:

Default:
image

With spacer_width set to 0.1:
image

Kandy can support it, since setting it manually like this works:
```kt
private val spacerWidthAes = Aes("spacer_width")
var PieContext.spacerWidth: Double?
    get() = (bindingCollector.settings[spacerWidthAes] as? NonPositionalSetting<Double>)?.value
    set(value) {
        bindingCollector.settings[spacerWidthAes] = NonPositionalSetting(spacerWidthAes, value)
    }
```

### PR DESCRIPTION
**Pull request by AndreiKingsley**
Several small API improvements 
Fixes  #322
Fixes #334
Fixes #385

### CODE REVIEW

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/layers/builders/aes/WithSpacerColor.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.jetbrains.kotlinx.kandy.util.color.Color
+import kotlin.reflect.KProperty
+
+/**
+ * Interface for managing the `spacerColor` aesthetic.
+ *
+ * Affects the color for spacers between sectors. Spacers are not applied
+ * to exploded sectors and to sides of adjacent sectors.
+ */
+@Suppress("INVISIBLE_MEMBER")
```
Comment (devcrocod) [inline kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/layers/builders/aes/WithSpacerColor.kt:20]: duplicate with file suppress

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/layers/builders/aes/WithSpacerWidth.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.jetbrains.kotlinx.kandy.letsplot.internal.SPACER_WIDTH
+import kotlin.reflect.KProperty
+
+/**
+ * Interface for managing the `spacerWidth` aesthetic.
+ *
+ * Affects the line width between sectors. Spacers are not applied
+ * to exploded sectors and to sides of adjacent sectors.
+ */
+@Suppress("INVISIBLE_MEMBER")
```
Comment (devcrocod) [inline kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/layers/builders/aes/WithSpacerWidth.kt:19]: duplicate

File: kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/layers/builders/aes/WithSpacerColor.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+public interface WithSpacerColor : WithAes {
+
+    /**
+     *
+     * @property stroke a numeric value that represents the spacer color.
+     */
+    public var spacerColor: Color?
```
Comment (devcrocod) [inline kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/layers/builders/aes/WithSpacerColor.kt:27]: Only setter, without mapping?

Comment (AndreiKingsley) [reply kandy-lets-plot/src/main/kotlin/org/jetbrains/kotlinx/kandy/letsplot/layers/builders/aes/WithSpacerColor.kt:27]: Yes, lets-plot doesn't support mapping on it.

**Review by devcrocod — APPROVED** [review]

**Pull request merged**